frontend/public/components/import-yaml-results.tsx

```diff
@@ -0,0 +1,299 @@
+  return column === IMPORT_COLS.STATUS
+    ? _.get(row.status, COLUMN_MAP[column])
+    : _.get(row.obj, COLUMN_MAP[column]);
+};
+
+const ImportYAMLResults: React.FC<ImportYAMLResultsPageProps> = ({
+  activeNamespace,
+  location,
+  models,
+}) => {
+  const { t } = useTranslation();
+  const [rows, setRows] = React.useState(
+    location.state.objs.map((obj) => ({
+      obj: { ...obj },
+      status: { creating: true, message: t('public~Creating') },
+    })),
+  );
+  const [sending, setSending] = React.useState(true);
+  const [errors, setErrors] = React.useState(false);
+  const [activeSortIndex, setActiveSortIndex] = React.useState(-1);
+  const [activeSortDirection, setActiveSortDirection] = React.useState(null);
+  const [ready, setReady] = React.useState(false);
+  React.useEffect(() => {
+    // Add a ready state so we're not rendering a flicker of a stale table (with the browser back button)
+    // before the history.goBack action occurs
+    if (location.state.done) {
+      return history.goBack();
+    }
+    setReady(true);
+    const requests = createRequests(location.state.objs, models);
+    Promise.allSettled(requests).then(() => {
+      history.replace({ state: { ...location.state, done: true } });
+      setSending(false);
+    });
+    requests.forEach((resourceRequest, index) => {
+      resourceRequest
+        .then((resourceObj) =>
+          setRows((prevRows) =>
+            // eslint-disable-next-line max-nested-callbacks
+            prevRows.map((resourceRow) =>
+              resourceObj?.metadata?.name === resourceRow.obj.metadata.name &&
+              resourceObj?.kind === resourceRow.obj.kind &&
+              resourceObj?.metadata?.namespace === resourceRow.obj.metadata.namespace
+                ? {
+                    obj: { ...resourceObj },
+                    status: { creating: false, message: t('public~Created') },
+                  }
+                : resourceRow,
+            ),
+          ),
+        )
+        .catch((error) => {
+          setErrors(true);
+          setRows((prevRows) =>
+            // eslint-disable-next-line max-nested-callbacks
+            prevRows.map((resourceRow) =>
+              location.state.objs[index]?.metadata?.name === resourceRow.obj.metadata.name &&
+              location.state.objs[index]?.kind === resourceRow.obj.kind &&
+              location.state.objs[index]?.metadata?.namespace === resourceRow.obj.metadata.namespace
+                ? {
+                    ...resourceRow,
+                    status: {
+                      creating: false,
+                      error: true,
+                      message: t('public~Error: {{error}}', { error: error.message }),
+                    },
+                  }
+                : resourceRow,
+            ),
+          );
+        });
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+  const onSort = (event, index, direction) => {
```

Comment:
I'm tempted to ask you to remove the sort. It seems to be adding a lot of complexity, and I don't see a lot of value to the user. I'd expect to rarely have more than 10 items.

@bmignano How important do you think sorting is?

Reply:
It looks like a lot of functions are missing types. Can you make sure we use types consistently throughout?

Reply:
@spadgett Agree sorting is probably not necessary here (I assumed we got it for free with the table). If it's adding lots of complexity, let's remove it!

Reply:
removed sorting